internal/certmonitor/certmonitor.go

```diff
@@ -0,0 +1,254 @@
+	monitor := &certMonitor{
+		informerfactory: informerFactory,
+		podInformer:     podInformer,
+		config:          config,
+		namespaceLister: namespaceLister,
+	}
+	return monitor, nil
+}
+
+// objectMatchesSelector func checks if object matches given label selector
+func objectMatchesSelector(obj runtime.Object, selector *metav1.LabelSelector) bool {
+	if selector == nil {
+		return true
+	}
+	labelselector, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return false
+	}
+
+	metaObj, ok := obj.(metav1.Object)
+	if !ok {
+		return false
+	}
+
+	return labelselector.Matches(labels.Set(metaObj.GetLabels()))
+
+}
+
+// secretMatches func checks if secret matches in the monitor config
+func (c *certMonitor) secretMatches(s *corev1.Secret, monitor MonitorConfig) bool {
+	if s == nil {
+		return false
+	}
+	if len(monitor.Secret.Name) > 0 && s.Name != monitor.Secret.Name {
+		return false
+	}
+	if len(monitor.Namespace.Name) > 0 && s.Namespace != monitor.Namespace.Name {
+		return false
+	}
+	if monitor.Secret.LabelSelector != nil && !objectMatchesSelector(s, monitor.Secret.LabelSelector) {
+		return false
+	}
+
+	if monitor.Namespace.LabelSelector != nil {
+		ns, err := c.namespaceLister.Get(s.Namespace)
+		if err != nil {
+			return false
+		}
+		if !objectMatchesSelector(ns, monitor.Secret.LabelSelector) {
+			return false
+		}
+	}
+	return true
+}
+
+// findCertsFromSecret func extracts, decodes, parses certifications from a secret, and displays in Prometheus
+func (c *certMonitor) findCertsFromSecret(secret *corev1.Secret) {
+	for dataKey, dataCert := range secret.Data {
+		certConv, err := base64.StdEncoding.DecodeString(string(dataCert))
+		if err != nil {
+			continue
+		}
+
+		pparse, _ := pem.Decode(certConv)
+		if pparse == nil {
+			continue
+		}
+
+		certss, err := x509.ParseCertificate(pparse.Bytes)
+		if err != nil {
+			continue
+		}
+
+		expiryTime := float64(certss.NotAfter.Unix())
+		certificatesExpiry.WithLabelValues(secret.Namespace, secret.Name, dataKey).Set(expiryTime)
+	}
+}
+
+// handleScretCreation func event handles new secret creations
+func (c *certMonitor) handleSecretCreation(obj interface{}) {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return
+	}
+	fmt.Printf("Handling Creation: %s,%s\n", secret.Namespace, secret.Name)
+	c.findCertsFromSecret(secret)
+}
+
+// handleSecretUpdate func event handles secret updates
+func (c *certMonitor) handleSecretUpdate(oldObj, newObj interface{}) {
+	oldsecret, ok := oldObj.(*corev1.Secret)
+	if !ok {
+		return
+	}
+
+	newsecret, ok := newObj.(*corev1.Secret)
+	if !ok {
+		return
+	}
+
+	if newObj == nil || oldObj == nil {
+		return
+	}
+	for oldKey := range oldsecret.Data {
+		if _, ok := newsecret.Data[oldKey]; !ok {
+			certificatesExpiry.DeletePartialMatch(prometheus.Labels{
```

Comment:
Can you add a comment like
```suggestion
		// oldKey does not exist in the new secret hence deleting the metric for that namespace/secret/key combination
			certificatesExpiry.DeletePartialMatch(prometheus.Labels{
```